worlds/rabi_ribi/locations.py

```diff
@@ -1,324 +1,393 @@
+    LocationName.atk_up_east_riverbank        : get_rabi_ribi_base_id() + 0x79,
+    LocationName.spike_barrier                : get_rabi_ribi_base_id() + 0x7A,
+    LocationName.frame_cancel                 : get_rabi_ribi_base_id() + 0x7B,
+    LocationName.hp_up_lab_slide_tunnel       : get_rabi_ribi_base_id() + 0x7C,
+    LocationName.mp_up_lab                    : get_rabi_ribi_base_id() + 0x7D,
+    LocationName.hp_up_riverbank              : get_rabi_ribi_base_id() + 0x7E,
+    LocationName.mp_up_evernight              : get_rabi_ribi_base_id() + 0x7F,
+    LocationName.hp_up_evernight              : get_rabi_ribi_base_id() + 0x80,
+    LocationName.hp_up_lab_pit                : get_rabi_ribi_base_id() + 0x81,
+    LocationName.sliding_powder               : get_rabi_ribi_base_id() + 0x82,
+    LocationName.atk_up_evernight_uprprc      : get_rabi_ribi_base_id() + 0x83,
+    LocationName.cashback                     : get_rabi_ribi_base_id() + 0x84,
+    LocationName.plus_necklace                : get_rabi_ribi_base_id() + 0x85,
+    LocationName.weaken                       : get_rabi_ribi_base_id() + 0x86,
+    LocationName.atk_up_lab_computer          : get_rabi_ribi_base_id() + 0x87,
+    LocationName.pack_up_south_evernight      : get_rabi_ribi_base_id() + 0x88,
+    LocationName.pack_up_north_evernight      : get_rabi_ribi_base_id() + 0x89,
+    LocationName.regen_up_evernight           : get_rabi_ribi_base_id() + 0x8A,
+    LocationName.atk_up_evernight             : get_rabi_ribi_base_id() + 0x8B,
+    LocationName.atk_up_east_lab              : get_rabi_ribi_base_id() + 0x8C,
+    LocationName.pack_up_lab                  : get_rabi_ribi_base_id() + 0x8D,
+    LocationName.hammer_roll                  : get_rabi_ribi_base_id() + 0x8E,
+}
+
+subterranean_area_locations = {
+    LocationName.hp_up_volcanic               : get_rabi_ribi_base_id() + 0x8F,
+    LocationName.fire_orb                     : get_rabi_ribi_base_id() + 0x90,
+    LocationName.pack_up_volcanic             : get_rabi_ribi_base_id() + 0x91,
+}
+
+system_interior_locations = {
+    LocationName.regen_up_cyberspace          : get_rabi_ribi_base_id() + 0x92,
+    LocationName.pack_up_cyberspace           : get_rabi_ribi_base_id() + 0x93,
+    LocationName.air_jump                     : get_rabi_ribi_base_id() + 0x94,
+    LocationName.hp_up_cyberspace             : get_rabi_ribi_base_id() + 0x95,
+    LocationName.atk_up_cyberspace            : get_rabi_ribi_base_id() + 0x96,
+    LocationName.mp_up_cyberspace             : get_rabi_ribi_base_id() + 0x97,
+}
+
+# Shufflable Gift Item Locations
+shufflable_gift_item_locations = {
+    LocationName.p_hairpin                    : get_rabi_ribi_base_id() + 0x98,
+    LocationName.speed_boost                  : get_rabi_ribi_base_id() + 0x99,
+    LocationName.bunny_strike                 : get_rabi_ribi_base_id() + 0x9A,
+}
+
+# Egg Locations
```

Comment:
Why separate the egg locations from the other item locations in the area?

Reply:
Honestly, it's not necessary for now, if ever. I was simply copying the items in order from `location_items.txt` , which separated them.

I guess it does mean we could do something like exclude easter egg locations if we ever wanted to. :)

Reply:
I forgot `locations_items.txt` did that. Makes sense 👍